include/envoy/network/udp_packet_writer_handler.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+
+#include "envoy/api/io_error.h"    // IoCallUint64Result
+#include "envoy/buffer/buffer.h"   // Buffer
+#include "envoy/network/address.h" // Address
+#include "envoy/network/socket.h"  // Socket
+#include "envoy/stats/scope.h"
+#include "envoy/stats/stats_macros.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * Max v6 packet size, excluding IP and UDP headers.
+ */
+static const uint64_t K_MAX_OUTGOING_PACKET_SIZE = 1452;
+
+/**
+ * UdpPacketWriterBuffer bundles a buffer and a function that
+ * releases it.
+ */
+struct UdpPacketWriterBuffer {
+  UdpPacketWriterBuffer() = default;
+  UdpPacketWriterBuffer(char* buffer, std::function<void(const char*)> release_buffer)
+      : buffer_(buffer), release_buffer_(std::move(release_buffer)) {}
+
+  char* buffer_ = nullptr;
+  std::function<void(const char*)> release_buffer_;
+};
+
+class UdpPacketWriter {
+public:
+  virtual ~UdpPacketWriter() = default;
+
+  /**
+   * @brief Sends a packet via given UDP socket with specific source address.
+   *
+   * @param buffer points to the buffer containing the packet
+   * @param local_ip is the source address to be used to send.
```

Comment:
What if it's null?

Reply:
In the cases when local_ip is set as NULL we pick the default network interface ip address for this field.

Reply:
Update the doc comment to explain this please